src/internal/group_api/mod.rs

```diff
@@ -351,6 +384,63 @@ pub(crate) fn get_group_keys<'a>(
     )
 }
 
+fn compare_users<T: Eq + std::hash::Hash + std::fmt::Debug, X>(
+    desired_users: &Vec<T>,
+    found_users: HashMap<T, X>,
+) -> IronOxideErr {
+    let desired_users_set: HashSet<&T> = HashSet::from_iter(desired_users);
+    let found_users_vec: Vec<T> = found_users.into_iter().map(|(x, _)| x).collect();
+    let found_users_set: HashSet<&T> = HashSet::from_iter(&found_users_vec);
+    let diff: Vec<&&T> = desired_users_set.difference(&found_users_set).collect();
+    IronOxideErr::UserDoesNotExist(format!("Failed to find the following users: {:?}", diff))
```

Comment:
Still using Debug here.

Reply:
I still haven't found an alternate way to display it. I tried implementing Display for UserId, but it didn't seem to make a difference (unless I did it wrong).
The output from Debug looks reasonable to me, so is there a concern with leaving it?

Reply:
I think debug is fine for this case. The error looks good, which is what we want.

Reply:
I did get this to work by making 
```rust
struct DisplayableVec<T: Display>(Vec<T>);
```

I think I'm going to back off this. The error itself is Display, so that's good. We have other places where we are using Debug to build up the error. If we want to address this, we address this as well.